client-network/src/network_ops/default_network_ops_client.rs

```diff
@@ -80,22 +112,25 @@ where
         attributes: StakedStateOpAttributes,
     ) -> Result<TxAux> {
         match from_address {
-            ExtendedAddr::BasicRedeem(ref redeem_address) => {
-                let transaction = UnbondTx::new(value, 0, attributes);
-                let public_key = self
-                    .wallet_client
-                    .find_public_key(name, passphrase, redeem_address)?
-                    .ok_or_else(|| Error::from(ErrorKind::AddressNotFound))?;
-
-                let private_key = self
-                    .wallet_client
-                    .private_key(passphrase, &public_key)?
-                    .ok_or_else(|| Error::from(ErrorKind::PrivateKeyNotFound))?;
-                let signature = private_key
-                    .sign(transaction.id())
-                    .map(StakedStateOpWitness::new)?;
-                Ok(TxAux::UnbondStakeTx(transaction, signature))
-            }
+            ExtendedAddr::BasicRedeem(ref redeem_address) => self
+                .get_staked_state_account(StakedStateAddress::BasicRedeem(*redeem_address))
+                .and_then(|account| {
+                    let nonce = account.nonce;
```

Comment:
this signing procedure is almost the same as for `WithdrawUnbondedTx`, so I guess the common parts could be abstracted out into a function?